### PR TITLE
added specs and validations for Role model

### PR DESCRIPTION
**Pull request by MothOnMars**
- added shoulda-matchers gem for one-line specs
- added FactoryGirl syntax methods

Partial fix for https://github.com/diaspora/diaspora/issues/4020

I also added a few validations that seemed to make sense for roles, and tweaked the `Role.is_admin?` method to return a boolean. Hopefully CI will catch any issues that these changes may cause, but please let me know if I should back out any of those changes to the model itself.

### CODE REVIEW

**Comment by jhass** [comment]
Mh, I'm not sure I like shoulda-matchers, it feels inconsistent with the expect syntax.

**Comment by MothOnMars** [comment]
@jhass, the one-line syntax used by shoulda-matchers is supported by RSpec, even when using the expect syntax: https://github.com/rspec/rspec-expectations/blob/master/Should.md#one-liners

I feel that the inconsistency is a small trade-off for being able to write a one-liner instead of multiple lines. It's a solid Thoughtbot gem with great support & [documentation](http://matchers.shoulda.io/). It's probably saved me hundreds of hours and thousands of lines of spec code over the years. Check it out...let me know what you think...and in the meantime, I'll start making Hound happier... :smile:

**Comment by jhass** [comment]
Mh, okay deal, we can try it for a while.

File: app/models/role.rb [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
-  scope :admins, -> { where(name: 'admin') }
+  validates :person, presence: true
+  validates :name, uniqueness: {scope: :person_id}
+  validates :name, inclusion: {in: %w(admin spotlight)}
+
+  scope :admins, -> { where(name: "admin") }
 
   def self.is_admin?(person)
-    find_by_person_id_and_name(person.id, 'admin')
+    self.exists?(person_id: person.id, name: "admin")
```
Comment (jhass) [inline app/models/role.rb]: Not sure why hound isn't complaining here, but the `self.` is unnecessary.

Comment (MothOnMars) [inline app/models/role.rb]: thanks - fixing

**Comment by jhass** [comment]
Thank you!

**Pull request merged**